### PR TITLE
mon,mgr: track mon devices like we do OSDs'

### PR DESCRIPTION
**Pull request by liewegas**
```
$ ceph device ls
DEVICE                                 HOST:DEV     DAEMONS                 LIFE EXPECTANCY 
INTEL_SSDPEDMD400G4_CVFT520200G7400BGN gnit:nvme0n1 mon.a osd.0 osd.1 osd.2                 
```
- also enables scraping of metrics from monitor daemons, just like osds.

### CODE REVIEW

**Comment by liewegas** [comment]
Now that ceph has a first-class device object, what I really want is 'ceph device ident <devid>' or similar to turn on ident or fault LEDs in enclosures...

File: src/pybind/mgr/devicehealth/module.py [hunk 1]
```diff
@@ -105,23 +105,21 @@ def handle_command(self, _, cmd):
         self.log.error("handle_command")
 
         if cmd['prefix'] == 'device query-daemon-health-metrics':
-            who = cmd.get('who', '')
-            if who[0:4] != 'osd.':
-                return -errno.EINVAL, '', 'not a valid <osd.NNN> id'
-            osd_id = who[4:]
+            (daemon_type, daemon_id) = cmd.get('who', '').split('.')
```
Comment (jcsp) [inline src/pybind/mgr/devicehealth/module.py]: Need some error handling here to give a friendly message on a 'who' that doesn't contain the period

File: src/pybind/mgr/devicehealth/module.py [hunk 2]
```diff
@@ -105,23 +105,21 @@ def handle_command(self, _, cmd):
...
+            if daemon_type not in ('osd', 'mon'):
+                return -errno.EINVAL, '', 'not a valid daemon name'
```
Comment (jcsp) [inline src/pybind/mgr/devicehealth/module.py]: Extend this string to advise that only osd and mon are allowed?  I can imagine occasional confused users being confused enough to try use this on an MDS.

**Comment by liewegas** [comment]
@jcsp updated

**Pull request merged**